static/js/admin/src/AddMenu.js

```diff
@@ -34,7 +33,7 @@ define('AddMenu', [ 'jquery', 'jquery-ui', 'underscore', 'CreateContainerForm',
 				items["addFile"] = { name : "Add File" };
 			}
 			
-			if (!workOrFile) {
+			if (this.container.type === "Collection") {
```

Comment:
the option for be available for Folders and Collections, so you can probably just use `folderOrCollection`